rmf_task_msgs/srv/SubmitTask.srv

```diff
@@ -0,0 +1,32 @@
+# Submit Task | POST service call
+
+# Whom that call this srv
+string requester
+
+# Desired start time of a task
+builtin_interfaces/Time start_time
+
+# Task type
+TaskType task_type
+
+# Task definitions
+Station station
+Loop loop
+Delivery delivery
+# Charge charge
+Clean clean
+# Patrol patrol
+
+# fleet selection evaluator
+uint8 evaluator
```

Comment:
I'm not sure if it makes sense to have a field for specifying the evaluation type. To me this carries several implications that I don't think are necessarily true:
1. The evaluation method might change arbitrarily during runtime
2. All possible evaluation methods that any system integrator might want to use must comply with one of these types (i.e. system integrators are not free to come up with a custom evaluation type)
3. A single enum is sufficient to fully define the evaluation method

I think a better approach in the long-run would be defining an abstract interface for every place where we evaluate task allocation quality, and have some kind of modular or plugin approach to allow system integrators to specify what evaluation method they want to use.

That being said, we'll be putting this out as a WIP, so we don't need to change this approach right now. Perhaps we can just open a ticket for it.

Reply:
I agree with this. Have logged an issue to follow up with https://github.com/osrf/rmf_core/issues/237